server/src/main/scala/viper/gobraserver/VerifierState.scala

```diff
@@ -32,15 +33,19 @@ object VerifierState {
 
   var verificationRunning: Int = 0
 
-  private val _jobQueue = mutable.Queue[(() => Program, () => BackTrackInfo, Long, VerifierConfig)]()
-  def jobQueue: mutable.Queue[(() => Program, () => BackTrackInfo, Long, VerifierConfig)] = _jobQueue
-
   private var _client: Option[IdeLanguageClient] = None
   def client: Option[IdeLanguageClient] = _client
 
   def setClient(client: IdeLanguageClient): Unit = {
     _client = Some(client)
   }
+
+  def submitVerificationJob(program: Program, backtrack: BackTrackInfo, startTime: Long, verifierConfig: VerifierConfig)(implicit executor: GobraExecutionContext): Unit = {
+    // simply call verify here without explicitly waiting on the result (or waiting for it in a runnable submitted
+    // to the thread pool - in this case an entire thread from the thread pool would be occupied waiting for it).
+    // executor.execute(() => GobraServer.verify(verifierConfig, program, backtrack, startTime))
+    GobraServer.verify(verifierConfig, program, backtrack, startTime)
```

Comment:
So this means that we limit the Gobra IDE to a single program being verified, right?

Reply:
Not really. `GobraServer.verify` returns a future with VerifierResult. As I understand it, we just start the verification of the Viper AST here and do not care about the VerifierResult. Instead, we use the message reporter to catch and forward the VerifierResult